server/channels/api4/channel.go

```diff
@@ -891,8 +905,11 @@ func getPublicChannelsByIdsForTeam(c *Context, w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	channelIds := model.ArrayFromJSON(r.Body)
-	if len(channelIds) == 0 {
+	channelIds, err := model.ArrayFromJSONLimited(r.Body, *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes)
+	if err != nil || len(channelIds) == 0 {
```

Comment:
This could be problematic if length is zero the error could be nil. Also we are handling the empty case below already.